pkg/planner/core/planbuilder.go

```diff
@@ -5419,6 +5446,97 @@ func (b *PlanBuilder) buildDDL(ctx context.Context, node ast.DDLNode) (base.Plan
 	return p, nil
 }
 
+// buildColsFromPlan builds the columns of the table from the logical plan
+// used in create table as select statement
+func buildColsFromPlan(v *ast.CreateTableStmt, logicalPlan base.LogicalPlan) error {
+	schema := logicalPlan.Schema()
+	names := logicalPlan.OutputNames()
+
+	// fill select columns,used in build sql when insert into table(v.SelectColumns...)  select ...
+	for _, name := range names {
+		v.SelectColumns = append(v.SelectColumns, &name.ColName)
+	}
+	if v.Cols == nil {
```

Comment:
follwing code is a little bit complicated and redundancy, could you try to refine it?